src/Sql/Sql/DistributedAvailabilityGroup/Cmdlet/FailoverAzureSqlManagedInstanceLink.cs

```diff
@@ -0,0 +1,201 @@
+        public SwitchParameter Force { get; set; }
+
+        /// <summary>
+        /// Get the entities from the service
+        /// </summary>
+        /// <returns>The list of entities</returns>
+        protected override IEnumerable<AzureSqlManagedInstanceLinkModel> GetEntity()
+        {
+            return new List<AzureSqlManagedInstanceLinkModel>() { ModelAdapter.GetManagedInstanceLink(ResourceGroupName, InstanceName, Name) };
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            switch (ParameterSetName)
+            {
+                case FailoverByNameParameterSet:
+                    // default case, we're getting RG, MI and MiLink names directly from args
+                    break;
+                case FailoverByParentObjectParameterSet:
+                    // we need to extract RG and MI name from the Instance object, MiLink name received directly from arg
+                    ResourceGroupName = InstanceObject.ResourceGroupName;
+                    InstanceName = InstanceObject.ManagedInstanceName;
+                    break;
+                case FailoverByInputObjectParameterSet:
+                    // we need to extract RG, MI and MiLink name directly from the MiLink object but replication mode can be either from the object or from arg
+                    ResourceGroupName = InputObject.ResourceGroupName;
+                    InstanceName = InputObject.InstanceName;
+                    Name = InputObject.Name;
+                    FailoverType = this.IsParameterBound(c => c.FailoverType) ? FailoverType : InputObject.ReplicationMode;
+                    break;
+                case FailoverByResourceIdParameterSet:
+                    // we need to derive RG, MI and MiLink name from resource id
+                    var resourceInfo = new ResourceIdentifier(ResourceId);
+                    ResourceGroupName = resourceInfo.ResourceGroupName;
+                    InstanceName = resourceInfo.ParentResource.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries).Last();
+                    Name = resourceInfo.ResourceName;
+                    break;
+                default:
+                    break;
+            }
+
+            // messages describing behavior with -WhatIf and -Confirm flags
+            if (ShouldProcess(
+                string.Format(CultureInfo.InvariantCulture, Properties.Resources.SetAzureSqlInstanceLinkDescription, ResourceGroupName, InstanceName, Name),
+                string.Format(CultureInfo.InvariantCulture, Properties.Resources.SetAzureSqlInstanceLinkWarning, ResourceGroupName, InstanceName, Name),
+                Properties.Resources.ShouldProcessCaption))
+            {
+                // message prompt requiring the customer to explicitly confirm the delete operation
+                if (!FailoverType.Equals("ForcedAllowDataLoss") || Force || (FailoverType.Equals("ForcedAllowDataLoss") &&
```

Comment:
I am not sure that I understood the question but this is the point: we have two failover types Planned and ForcedAllowDataLoss which do different things on backend side. If user wants to perform ForcedAllowDataLoss failover we need to ask for confirmation because data loss can happen. When performing Planned failover data loss can't happen and that is why we are not asking for confirmation.

Reply:
Yes, just wanted to confirm you have different logic on the server side.